tests/FSharp.Compiler.ComponentTests/Language/DotLambdaTests.fs

```diff
@@ -290,4 +290,31 @@ let _ = asQ.Select _.Length
 """
     |> withLangVersion80
     |> typecheck
-    |> shouldSucceed
\ No newline at end of file
+    |> shouldSucceed
+
+[<Fact>]
+    let ``Error when property has same name as DU case`` () =
+        Fsx """
+type MyId =
+    | IdA of int
+    | IdB of string
+
+    member this.IdA =
+        match this with
+        | IdA x -> Some x
+        | _ -> None
+        
+    member this.IdX =
+        match this with
+        | IdB x -> Some x
+        | _ -> None
+
+let onlyIdX (ids: MyId list) = ids |> List.choose _.IdX
+let onlyIdA (ids: MyId list) = ids |> List.choose _.IdA
```

Comment:
@psfinaki This now raises an error message if you accessing to a member that has the same name as the union case. reusing `FS0023`

`(Error 23, Line 17, Col 51, Line 17, Col 56, "The member 'IdA' can not be defined because the name 'IdA' clashes with the union case 'IdA' in this type or module")`

I propose we update the error message so it can be used in both cases ie
  - `member` is defined 
  -  `member` is used

Avoiding having to create another error number and message. Maybe

- `The member 'IdA' can not be used because 'IdA' clashes with the union case 'IdA' in this type or module`

- `The name 'IdA' can not be used because 'IdA' clashes with the union case 'IdA' in this type or module`

- `The member 'IdA' can not be defined or accessed because the name 'IdA' clashes with the union case 'IdA' in this type or module`

cc @vzarytovskii

Reply:
Hmm but why not having both errors here? As in,
```fsharp
type MyId =
    | IdA of int
    | IdB of string

    member this.IdA =
        match this with
        | IdA x -> Some x
        | _ -> None
 
let onlyIdA (ids: MyId list) = ids |> List.choose _.IdA
```

...would produce both 
`FS0023 The member 'IdA' can not be defined because the name 'IdA' clashes with the union case 'IdA' in this type or module` and 
`FS0812	The syntax 'expr.id' may only be used with record labels, properties and fields`,
as if they were independent (which they kind of are)?

Reply:
The issue is that the existing logic uses `error(Error(..))` which prevents the analysis to continue so even if we try to add both they would cancel each other. Ideally it would be good to change to use `errorR`  but will require rewriting of pat of the analysis. The best we can do for now is to reuse `FS0023`

Reply:
I mean I probably misunderstand something here - why cannot/shouldn't we fix the error recovery in this case instead?

Reply:
It is not that we can't. But it will required updating/splitting the upstream analysis to make it work as `error` returns `exn -> 'T` and `errorR` returns `exn -> unit`.

Reply:
But if you feel adventurous we can maybe pair on this, maybe in an AmplifyingF# session :)

Reply:
Context: This is the relevant piece of code that will needs updating https://github.com/dotnet/fsharp/blob/f29ce0aef9b70bfef03ef7466ec823e449bb864b/src/Compiler/Checking/CheckExpressions.fs#L9270C5-L9270C21

Reply:
Well I just believe this can be an opportunity to fix the underlying issue rather than patch the symptome. But that's just my opinion :eyes:

Otherwise, "The member 'IdA' can not be defined or **accessed**" is something I would pick out of the options.

Reply:
Well at the very least I would give a try and report some progress :)

Reply:
@psfinaki Im revisiting this here https://github.com/dotnet/fsharp/pull/17088